js/index.js

```diff
@@ -1,10 +1,77 @@
 const isEqual = require("lodash.isequal");
 const expects = require("./expects.js");
 
+function getAllIndexes(arr, val) {
+  var indexes = [], i;
+  for (i = 0; i < arr.length; i++)
+    if (arr[i] === val)
+      indexes.push(i);
+  return indexes;
+}
+
+function checkVertical(puzzle, y, x, word) {
+  for (let i = 0; i < word.length; i++) {
+    if (word[i] !== puzzle[y][x]) {
+      return 0
+    }
+    y++
+  }
+  return 1
+}
+
+function checkDiagonal(puzzle, y, x, word) {
+  let n = 0
+  //check up diagonal
+  let flagup = 1
+  //check down diagonal
+  let flagdown = 1
+  for (let i = 0; i < word.length; i++) {
+    if (y + n >= puzzle.length || x + n >= puzzle[y + n].length || word[i] !== puzzle[y + n][x] || flagdown) {
+      flagdown = 0
+    }
+
+    if (y - n < 0 || x + n >= puzzle.length || word[i] !== puzzle[y - n][x + n]) {
+      flagup = 0
+    }
+    n++;
+  }
+  //to know which end index to compute
+  return flagup ? 1 : (flagdown ? -1 : 0);
+}
 function wordSearch(puzzle, ...words) {
-  // Add your code here
+  const obj = {}
+
+  words.forEach(w => {
+    obj[w] = [];
+    for (let n = 0; n < puzzle.length; n++) {
+      const indices = getAllIndexes(puzzle[n], w[0])
+      indices.forEach(i => {
+        //horizontal
+        if (i + w.length <= puzzle[n].length) {
+          if (puzzle[n].slice(i, w.length + i).join("") === w) {
```

Comment:
Very interesting approach. I like it. No one else had the same approach as this. A simple clear approach